cpp/src/unary/cast_ops.cu

```diff
@@ -236,7 +269,56 @@ struct dispatch_unary_cast_to {
     return output;
   }
 
+  template <
+    typename TargetT,
+    typename SourceT                                                  = _SourceT,
+    typename std::enable_if_t<cudf::is_fixed_point<SourceT>() && cudf::is_fixed_point<TargetT>() &&
+                              std::is_same<SourceT, TargetT>::value>* = nullptr>
+  std::unique_ptr<column> operator()(data_type type,
+                                     rmm::mr::device_memory_resource* mr,
+                                     cudaStream_t stream)
+  {
+    if (input.type() == type) return std::make_unique<column>(input);  // TODO add test for this
+
+    return detail::rescale<TargetT>(input, type, mr, stream);
+  }
+
+  template <
+    typename TargetT,
+    typename SourceT                                                      = _SourceT,
+    typename std::enable_if_t<cudf::is_fixed_point<SourceT>() && cudf::is_fixed_point<TargetT>() &&
+                              not std::is_same<SourceT, TargetT>::value>* = nullptr>
+  std::unique_ptr<column> operator()(data_type type,
+                                     rmm::mr::device_memory_resource* mr,
+                                     cudaStream_t stream)
+  {
+    using namespace numeric;
+
+    auto const size = input.size();
+    auto temporary =
+      std::make_unique<column>(cudf::data_type{type.id(), input.type().scale()},
+                               size,
+                               rmm::device_buffer{size * cudf::size_of(type), stream, mr},
+                               copy_bitmask(input, stream, mr),
+                               input.null_count());
+
+    using SourceDeviceT = device_storage_type_t<SourceT>;
+    using TargetDeviceT = device_storage_type_t<TargetT>;
+
+    mutable_column_view output_mutable = *temporary;
+
+    thrust::transform(rmm::exec_policy(stream)->on(stream),
+                      input.begin<SourceDeviceT>(),
+                      input.end<SourceDeviceT>(),
+                      output_mutable.begin<TargetDeviceT>(),
+                      device_cast<SourceDeviceT, TargetDeviceT>{});
+
+    // clearly there is a more efficient way to do this, can optimize in the future
```

Comment:
Seems like it would be useful to binary_ops have an iterator detail interface that you could send a transform iterator into.

Reply:
That would definitely be useful here but would seem atypical for a libcudf API. Not sure if worth adding to this PR